packages/sync-actions/src/product-types.js

```diff
@@ -0,0 +1,53 @@
+/* @flow */
+import flatten from 'lodash.flatten'
+import type { SyncAction, UpdateAction, ActionGroup } from '../../../types/sdk'
+import createBuildActions from './utils/create-build-actions'
+import createMapActionGroup from './utils/create-map-action-group'
+import * as productTypeActions from './product-types-actions'
+import * as diffpatcher from './utils/diffpatcher'
+import findMatchingPairs from './utils/find-matching-pairs'
+
+const actionGroups = ['base', 'attributes']
+
+function createProductTypeMapActions(
+  mapActionGroup: (
+    type: string,
+    fn: () => Array<UpdateAction>
+  ) => Array<UpdateAction>
+): (diff: Object, next: Object, previous: Object) => Array<UpdateAction> {
+  return function doMapActions(
+    diff: Object,
+    next: Object,
+    previous: Object
```

Comment:
What're the `next` and `previous` parameters? And what's the rationale behind using them?

Reply:
I just don't like `newObj` and `oldObj`. please don't make me go back those namings

to me `new` vs `old` sounds to me that they are objects without the same reference.
at least with `next` vs `previous`, we referr to the same entity of _potentially different versions_